### ISSUE TITLE
how to type Pagination with @gitbeaker/rest

### ISSUE THREAD

**Issue by xianengzhi** [jdalrymple/gitbeaker#3279]
**Description**
in past,
i can type `{ page = 1, perPage = 10 }` with `Types.PaginatedRequestOptions` in `@gitbeaker/node`.
```typescript
  async search(
    projectName: string,
    { page = 1, perPage = 10 }: Types.PaginatedRequestOptions,
  ) {
    const data = await this.gitlab.Projects.search(projectName, {
      page,
      perPage,
      showExpanded: true,
    }).catch((err) => {
      console.log(err);
    });

    return data;
  }
```
now,
how can i type it with @gitbeaker/rest?
there is only 3 RequestOptions,but none can suit this situation.
```typescript
interface KeysetPaginationRequestOptions {
    orderBy: string;
    sort: 'asc' | 'dec';
}
interface OffsetPaginationRequestOptions {
    page?: number | string;
    maxPages?: number;
}
interface BasePaginationRequestOptions<P extends PaginationTypes | void> {
    pagination?: P;
    perPage?: number | string;
}
```

**Checklist**
- [ ] I have checked that this is not a duplicate issue.
- [ ] I have read the documentation.

**Comment by jdalrymple** [comment]
A few ways to do this:

1. Specify the type of pagination explicitly: `PaginationRequestOptions<'keyset'> ` or `PaginationRequestOptions<'offset'>`
2. Use the Wrapper without explicit assignment which includes both: `PaginationRequestOptions` 

The last option is included in the build currently being deployed. These latest changes also allow you to explicitly set keyset pagination without requiring properties by default.

**Comment by jdalrymple** [comment]
I've reached my max CI minutes for the month -_-, this will deploy on June 1

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`38.10.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/38.10.0) :rocket:

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`38.10.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/38.10.0) :rocket:
